### PR TITLE
Implementation of logLevel

### LINKED ISSUE #6: Add `logLevel` option
I'm thinking it might be useful to have a `logLevel` option, so you could choose what level to log based on the environment. I think a good default would be to set it to `log` (info) in development and `warn` in production, so it doesn't output `log` messages in production.

There should also be a way to set it for all logger instances, maybe:

```js
const logger = require('timber');

logger.setDefaults({
	logLevel: 'warn'
});
```

@acheronfail @lukechilds Thoughts?

### PR DESCRIPTION
**Pull request by acheronfail**
An initial implementation of `logLevel` (closes #6).

I left some notes on the changes, please review those when you can.

PS: I also added `logger.streamWarn` as well, which I think should be there since we already had `streamLog` and `streamError` etc.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -142,6 +171,31 @@ class Timber {
...
+		const defaults = {};
+		if (is.renderer) {
+			Object.assign(defaults, electron.remote.getGlobal(defaultsNameSpace) || {});
+		} else {
+			Object.assign(defaults, global[defaultsNameSpace]);
+		}
+
+		return defaults;
+	}
+
```
Comment (acheronfail) [inline index.js]: I thought might be handy to have for debugging purposes.
This method returns a copy of the defaults, so users don't accidentally mutate them.

File: test/fixture/test.js [hunk 2]
```diff
@@ -60,3 +60,49 @@ test.serial('toggle loggers', async t => {
...
+			t.regex(mainLogs[4], /renderer › error/);
+
+			let rendererLogs = await app.client.getRenderProcessLogs();
+			rendererLogs = cleanLogs(rendererLogs.map(x => x.message));
+			t.is(rendererLogs.length, 1);
+			t.regex(rendererLogs[0], /error/);
+			
+			resolve();
+		}, 1000);
+	});
```
Comment (acheronfail) [inline test/fixture/test.js]: Testing this turned out to be trickier than I thought.

I found that I couldn't reliably get the right logs unless I waited here. This definitely feels like a hack - so please tell me if I missed anything obvious!

File: index.js [hunk 3]
```diff
@@ -142,6 +171,31 @@ class Timber {
...
+	setDefaults(newDefaults = {}) {
+		if (is.renderer) {
+			throw new Error('setDefaults can only be called from the main process');
+		}
+
+		// We don't want the "name" property being set as a default.
+		delete newDefaults.name;
+		if ({}.hasOwnProperty.call(newDefaults, 'logLevel')) {
```
Comment (sindresorhus) [inline index.js]: Use `Reflect.has()` instead

File: test/fixture/test.js (hunk 2 above)
Comment (sindresorhus) [reply test/fixture/test.js]: Spectron is unreliable, so I think this is fine, but use https://github.com/sindresorhus/delay with `await` here to make it more readable.

File: index.js [hunk 4]
```diff
@@ -142,6 +171,31 @@ class Timber {
...
+	getDefaults() {
+		const defaults = {};
+		if (is.renderer) {
+			Object.assign(defaults, electron.remote.getGlobal(defaultsNameSpace) || {});
+		} else {
+			Object.assign(defaults, global[defaultsNameSpace]);
+		}
+
+		return defaults;
+	}
```
Comment (sindresorhus) [inline index.js]: Could be simplified:

```js
getDefaults() {
	const defaults = is.renderer ? electron.remote.getGlobal(defaultsNameSpace) : global[defaultsNameSpace];
	return Object.assign({}, defaults);
}
```

File: readme.md [hunk 5]
```diff
@@ -109,6 +113,25 @@ Type `RegExp`
 
 Ignore lines matching the given regex.
 
+##### logLevel
+
+Type: `string`
+
+May be `"info"` (log everything), `"warn"` (log warnings and errors) or `"error"` (log errors only). Defaults to `"info"` during development and `"warn"` in production.
```
Comment (sindresorhus) [inline readme.md]: Drop the `"` quotes. The type is already specified.

File: index.js [hunk 6]
```diff
@@ -27,12 +41,16 @@ class Timber {
 		}
 	}
 
+	get options() {
```
Comment (sindresorhus) [inline index.js]: Let prefix this with `_`. I don't see why anyone would need this. We can rather expose it if there's an actual need.

Comment (acheronfail) [reply index.js]: The property `this._options` is already taken by the actual object on the instance (not the getter method), that's why I originally left it as `this.options`. Should we add an extra `_` to both? Or rename one/both?

Comment (sindresorhus) [reply index.js]: You could rename the current `this._options` to `this._initialOptions`

**Comment by sindresorhus** [comment]
Thanks for your patience, @acheronfail. This looks excellent. 🙌✨

**Comment by sindresorhus** [comment]
@acheronfail Any idea why tests fail when it's merged to master? Tests were passing on the PR: https://travis-ci.org/sindresorhus/electron-timber/builds/358724775 Weird

**Comment by acheronfail** [comment]
@sindresorhus no... I just pulled the latest master and ran the tests on my machine and they were all green. That error message on Travis looks dodgy anyway.

**Comment by sindresorhus** [comment]
I already tried rerunning Travis twice, same thing, but I see now that it's green with the `0.4.0` commit. Definitely something flaky going on...

**Pull request merged**